### ISSUE TITLE
JDK8 MacOS : JVMSHRC659E An error has occurred while opening shared memory

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#13016]
Failure link
------------

From an internal build `Test_openjdk8_j9_extended.functional_x86-64_mac_testList_1/50/` (`osxrt2`):
```
openjdk version "1.8.0_302"
OpenJDK Runtime Environment (build 1.8.0_302-b05)
Eclipse OpenJ9 VM (build master-0be9a6f3f, JRE 1.8.0 Mac OS X amd64-64-Bit Compressed References 20210618_157 (JIT enabled, AOT enabled)
OpenJ9   - 0be9a6f3f
OMR      - b7eb96336
JCL      - be819230 based on jdk8u302-b05)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=upstream&amp;TARGET=testSCCacheManagement_0&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk8_j9_extended.functional_x86-64_mac&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&amp;TEST_PARALLELIZATION_PARAMS=&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_mac&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;LABEL=&amp;TEST_OPTIONS_PARAMS=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-mac-x64-openj9%2F157%2FOpenJDK8U-debugimage_x64_mac_openj9_2021-06-19-02-30.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-mac-x64-openj9%2F157%2FOpenJDK8U-jre_x64_mac_openj9_2021-06-19-02-30.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-mac-x64-openj9%2F157%2FOpenJDK8U-jdk_x64_mac_openj9_2021-06-19-02-30.tar.gz+https%3A%2F%2Fna.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk8u%2Fjdk8u-mac-x64-openj9%2F157%2FOpenJDK8U-testimage_x64_mac_openj9_2021-06-19-02-30.tar.gz&amp;BUILD_IDENTIFIER=&amp;NON_AQA_TEST_REPOS_HELP_TEXT=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=true&amp;NON_AQA_TEST_REPOS=&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;TEST_REPO_PARAMS=&amp;TEST_SELECTION_PARAMS=&amp;TEST_PARALLELIZATION_PARAMS_HELP_TEXT=&amp;JDK_SELECTION_PARAMS=&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=&amp;JDK_VERSION=8&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk8&amp;PLATFORM_AND_MACHINE_HELP_TEXT=&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;PLATFORM_AND_MACHINE=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;POST_RUN_PARAMS_HELP_TEXT=&amp;UPSTREAM_TEST_JOB_NUMBER=73&amp;DOCKERIMAGE_TAG=&amp;JDK_SELECTION_PARAMS_HELP_TEXT=&amp;JDK_IMPL=openj9&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;TKG_SHA=&amp;TEST_SELECTION_PARAMS_HELP_TEXT=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=4e18ffe7-b1b1-4272-9979-99769b68bcc2&amp;OPENJDK_SHA=&amp;NUM_MACHINES=&amp;TRSS_URL=http%3A%2F%2Ftrss1.fyre.ibm.com&amp;BUILD_LIST=functional&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TEST_REPO_PARAMS_HELP_TEXT=&amp;POST_RUN_PARAMS=&amp;TIME_LIMIT=20&amp;TEST_OPTIONS_PARAMS_HELP_TEXT=&amp;JVM_OPTIONS=&amp;PARALLEL=None)

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2021-06-19T03:43:42.824Z] Running test testSCCacheManagement_0 ...
[2021-06-19T03:43:42.824Z] ===============================================
[2021-06-19T03:43:42.824Z] testSCCacheManagement_0 Start Time: Fri Jun 18 23:43:40 2021 Epoch Time (ms): 1624074220215
[2021-06-19T03:43:42.824Z] "/Users/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_mac_testList_1/openjdkbinary/j2sdk-image/Contents/Home/bin/..//bin/java" -Xshareclasses:destroyAll; "/Users/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_mac_testList_1/openjdkbinary/j2sdk-image/Contents/Home/bin/..//bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2021-06-19T03:43:42.824Z] 
[2021-06-19T03:43:42.824Z] Attempting to destroy all caches in cacheDir /Users/jenkins/javasharedresources/
[2021-06-19T03:43:42.824Z] 
[2021-06-19T03:43:42.824Z] JVMSHRC806I Compressed references persistent shared cache "sharedcc_jenkins" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
[2021-06-19T03:43:42.824Z] JVMSHRC807I Non-compressed references persistent shared cache "sharedcc_jenkins" has been destroyed. Use option -Xcompressedrefs if you want to destroy a compressed references cache.
[2021-06-19T03:43:42.824Z] JVMSHRC005I No shared class caches available
[2021-06-19T03:43:42.824Z] cache cleanup done
[2021-06-19T03:43:42.824Z] variation: NoOptions
[2021-06-19T03:43:42.824Z] JVM_OPTIONS:  

[2021-06-19T03:43:42.825Z] java_exe =	/Users/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_mac_testList_1/openjdkbinary/j2sdk-image/Contents/Home/bin/..//bin/java  
[2021-06-19T03:43:42.825Z] cmd.runProgramSimpleApp =	null
[2021-06-19T03:43:42.825Z] .Running  :testCacheCreationListingDestroying01  (Test suite : TestOptionsDefault)
[2021-06-19T03:43:42.825Z] 
[2021-06-19T03:43:42.825Z] [-Xshareclasses persistent cache disabled]
[2021-06-19T03:43:42.825Z] [-Xshareclasses verbose output enabled]
[2021-06-19T03:43:42.825Z] JVMSHRC659E An error has occurred while opening shared memory
[2021-06-19T03:43:42.825Z] JVMSHRC336E Port layer error code = -174
[2021-06-19T03:43:42.825Z] JVMSHRC337E Platform error message: Invalid argument
[2021-06-19T03:43:42.825Z] JVMSHRC026E Cannot create cache of requested size: Please check your SHMMAX and SHMMIN settings
[2021-06-19T03:43:42.825Z] JVMSHRC663I Error recovery: destroyed semaphore set with id=65537 associated with shared class cache.
[2021-06-19T03:43:42.825Z] JVMSHRC840E Failed to start up the shared cache.
[2021-06-19T03:43:42.825Z] JVMJ9VM015W Initialization error for library j9shr29(11): JVMJ9VM009E J9VMDllMain failed
[2021-06-19T03:43:42.825Z] Error: Could not create the Java Virtual Machine.
[2021-06-19T03:43:42.825Z] Error: A fatal exception has occurred. Program will exit.
```

There are a few similar failures at same machine, so it seems a machine specific issue.
fyi @hangshao0

**Comment by pshipton** [comment]
@jdekonin @AdamBrousseau looks like a machine configuration issue.

**Comment by hangshao0** [comment]
The error message is:

> JVMSHRC026E Cannot create cache of requested size: Please check your SHMMAX and SHMMIN settings

**Comment by jdekonin** [comment]
That machine was disabled for upgrade to 11.4 last week, it should not have been enabled.  I'll rerun that pipeline as part of bringing that system back online.

**Comment by a7ehuo** [comment]
I'm seeing the same error with `testSCCacheManagement_0`  in [my personal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk11_j9_extended.functional_x86-64_mac_Personal/112/) 

machine: `osxrt4`
```
****************************** MACHINE INFO ******************************
uname : Darwin osxrt4 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
cpuCores : 4
sysArch : x86_64
procArch : i386
sysOS : Darwin
```

```
[-Xshareclasses verbose output enabled]
JVMSHRC659E An error has occurred while opening shared memory
JVMSHRC336E Port layer error code = -174
JVMSHRC337E Platform error message: Invalid argument
JVMSHRC026E Cannot create cache of requested size: Please check your SHMMAX and SHMMIN settings
JVMSHRC663I Error recovery: destroyed semaphore set with id=69926913 associated with shared class cache.
JVMSHRC840E Failed to start up the shared cache.
JVMJ9VM015W Initialization error for library j9shr29(11): JVMJ9VM009E J9VMDllMain failed
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

**Comment by pshipton** [comment]
@jdekonin pls see the previous comment.

**Comment by jdekonin** [comment]
These are the settings (which I needed to re-add) after the upgrade to 11.4.  Looks like these values are not used so I did a bit of searching and the method of setting changes after Catalina.

```
osxrt4:~ jenkins$ sw_vers
ProductName:	macOS
ProductVersion:	11.4
BuildVersion:	20F71
osxrt4:~ jenkins$ cat /etc/sysctl.conf 
kern.sysv.shmmax=125839605760
kern.sysv.shmall=30722560
kern.coredump=1
```

after creating a file in /Library/LaunchDaemons/ and using launchctl I have it working on reboot.

```
osxrt6:~ jenkins$ sysctl -A kern.sysv.shmall
kern.sysv.shmall: 30722560
osxrt6:~ jenkins$ sysctl -A kern.sysv.shmmax
kern.sysv.shmmax: 125839605760
```

I'll update instructions for new installs and go fix the existing system that I upgraded to 11.4 in the past few days.

**Comment by jdekonin** [comment]
All internal systems updated as per issue 5636.
